src/coreComponents/common/DataTypes.hpp

```diff
@@ -78,7 +78,7 @@ namespace geos
 using size_t      = std::size_t;
 
 /// Signed integer type.
-using integer     = std::int32_t;
+using integer     = int;
```

Comment:
FYI, @victorapm @wrtobin @CusiniM @dkachuma 

The issue in #3694 and #3693 came up again when I updated our cuda12 versions on the CI. This was the problem. The compiler was trying to do a cast in a template parameter list, and apparently it is becoming stricter about such things....or it is a bug.

Reply:
Interesting... But `int` and `std::int32_t` have the same size, right? Still puzzling... but thanks Randy!